test/test_buffer.py

```diff
@@ -173,9 +174,48 @@ def test_get_exceptions():
 def test_contains():
     ok(vim.current.buffer in vim.buffers)
 
+
 @with_setup(setup=cleanup)
 def test_set_items_for_range():
     vim.current.buffer[:] = ['a', 'b', 'c', 'd', 'e']
     r = vim.current.buffer.range(1, 3)
     r[1:3] = ['foo']*3
     eq(vim.current.buffer[:], ['a', 'foo', 'foo', 'foo', 'd', 'e'])
+
+
+@with_setup(setup=cleanup)
+def test_region():
+    reference_buffer = ['foo1', 'foo2', 'foo3', 'foo4', 'foo5', 'foo6']
+
+    vim.current.buffer[:] = reference_buffer
+    eq(vim.current.buffer[2:5], ['foo3', 'foo4', 'foo5'])
+
+    r = vim.current.buffer.range(3, 5)
+    eq(r[:], ['foo3', 'foo4', 'foo5'])
+    eq(r[0], 'foo3')
+
+    # reading
+    r = Region(vim.current.buffer, 2, 5)
+    eq(r[:], ['foo2', 'foo3', 'foo4', 'foo5'])
+
+    r = Region(vim.current.buffer, 2, 5, partials=(1, 3))
+    eq(r[:], ['oo']*4)
+
+    r = Region(vim.current.buffer, 2, 4, partials=[(1, 3), (1, 3), (0, 3)])
+    eq(r[:], ['oo']*2 + ['foo'])
+
+    # writing
+    vim.current.buffer[:] = reference_buffer
+    r = Region(vim.current.buffer, 2, 4, partials=[(1, 3), (1, 3), (0, 3)])
+    r[:] = ['bar']*3
```

Comment:
I would strongly prefer explicit functions rather than cute list operations. I don't know if it's pythonic. This is an RPC API, not pandas or whatever. Though I also find pandas far too magical.

@bfredl uses python more than I ,  would like to hear his opinion.

Reply:
To be honest until extmarks (and api or lua functions to manipulate charwise ranges in an efficient and atomic way) are available this is a pretty weak abstraction, "cute" list operations (no matter what one thinks about them) are more or less what the class offers.

Reply:
@justinmk 
Regarding the list-operatons I tried to keep as close as possible to the Buffer- and Range-classes, that's where the idea for this kind of implementation came from.
Regarding the API: which API is this wrapping? The code footprint would actually much decrease if there is an API I overlooked giving me the current selection that I can use.

@justinmk @bfredl 
So we proceed with the list operators like Buffer and Range use it for the Region-abstraction (until there is something better)?

Reply:
Ok, fair enough